src/lib/accessibility/FocusZone/FocusTrapZone.tsx

```diff
@@ -0,0 +1,306 @@
+    FocusTrapZone._focusStack.push(this)
+    const { disableFirstFocus = false } = this.props
+
+    this.windowElement = getWindow(this._root.current)
+    this._previouslyFocusedElementOutsideTrapZone = this._getPreviouslyFocusedElementOutsideTrapZone()
+
+    if (
+      !this._root.current.contains(this._previouslyFocusedElementOutsideTrapZone) &&
+      !disableFirstFocus
+    ) {
+      this.focus()
+    }
+
+    this._hideContentFromAccessibilityTree()
+    this._subscribeToEvents()
+  }
+
+  public render(): JSX.Element {
+    const { className, ariaLabelledBy } = this.props
+    const rest = getUnhandledProps(
+      { handledProps: [..._.keys(FocusTrapZone.propTypes)] },
+      this.props,
+    )
+    const ElementType = getElementType({ defaultProps: FocusTrapZone.defaultProps }, this.props)
+
+    return (
+      <ElementType
+        {...rest}
+        className={className}
+        ref={this.createRef}
+        aria-labelledby={ariaLabelledBy}
+        onKeyDown={this._onKeyboardHandler}
+        onFocusCapture={this._onFocusCapture}
+      >
+        {this.props.children}
+      </ElementType>
+    )
+  }
+
+  public componentWillUnmount(): void {
+    const { ignoreExternalFocusing } = this.props
+
+    FocusTrapZone._focusStack = FocusTrapZone._focusStack.filter((value: FocusTrapZone) => {
+      return this !== value
+    })
+
+    const activeElement = document.activeElement as HTMLElement
+    if (
+      !ignoreExternalFocusing &&
+      this._previouslyFocusedElementOutsideTrapZone &&
+      (this._root.current.contains(activeElement) || activeElement === document.body)
+    ) {
+      focusAsync(this._previouslyFocusedElementOutsideTrapZone)
+    }
+
+    this._unsubscribeFromEvents()
+    const lastActiveFocusTrap = FocusTrapZone._focusStack[FocusTrapZone._focusStack.length - 1]
+
+    if (!FocusTrapZone._focusStack.length) {
+      this._showContentInAccessibilityTree()
+    } else if (
+      lastActiveFocusTrap._root.current &&
+      lastActiveFocusTrap._root.current.hasAttribute(HIDDEN_FROM_ACC_TREE)
+    ) {
+      lastActiveFocusTrap._root.current.removeAttribute(HIDDEN_FROM_ACC_TREE)
+      lastActiveFocusTrap._root.current.removeAttribute('aria-hidden')
+    }
+  }
+
+  public focus() {
+    if (!this._root.current) return
+    const { focusPreviouslyFocusedInnerElement, firstFocusableSelector } = this.props
+
+    if (
+      focusPreviouslyFocusedInnerElement &&
+      this._previouslyFocusedElementInTrapZone &&
+      this._root.current.contains(this._previouslyFocusedElementInTrapZone)
+    ) {
+      // focus on the last item that had focus in the zone before we left the zone
+      focusAsync(this._previouslyFocusedElementInTrapZone)
+      return
+    }
+
+    const focusSelector =
+      typeof firstFocusableSelector === 'string'
+        ? firstFocusableSelector
+        : firstFocusableSelector && firstFocusableSelector()
+
+    let _firstFocusableChild
+
+    if (focusSelector) {
```

Comment:
this could be simplified - as an additional benefit we are eliminating mutable `let` semantics for the variable

```js
const firstFocusableChild = focusSelector
  ? this._root.current.querySelector('.' + focusSelector)
  : getNextElement(
        this._root.current,
        this._root.current.firstChild as HTMLElement,
        true,
        false,
        false,
        true,
      )
    }
```